$OpenBSD: patch-program_steps_mail_get_inc,v 1.1 2013/01/03 20:05:15 sthen Exp $

4163511314  Workaround MSIE<=8 bug; content-disposition:inline ignored
if filename contains "attachment"

--- program/steps/mail/get.inc.orig	Wed Nov 14 20:30:28 2012
+++ program/steps/mail/get.inc	Thu Jan  3 19:45:36 2013
@@ -150,6 +150,13 @@ else if (strlen($pid = get_input_value('_part', RCUBE_
 
       $disposition = !empty($plugin['download']) ? 'attachment' : 'inline';
 
+      // Workaround for nasty IE bug (#1488844)
+      // If Content-Disposition header contains string "attachment" e.g. in filename
+      // IE handles data as attachment not inline
+      if ($disposition == 'inline' && $browser->ie && $browser->ver < 9) {
+        $filename = str_ireplace('attachment', 'attach', $filename);
+      }
+
       header("Content-Disposition: $disposition; filename=\"$filename\"");
 
       // do content filtering to avoid XSS through fake images
